module/VuFind/src/VuFind/RecordDriver/DefaultRecord.php

```diff
@@ -1722,4 +1722,80 @@ public function getCoordinateLabels()
         return isset($this->fields['long_lat_label'])
             ? $this->fields['long_lat_label'] : [];
     }
+
+    /**
+     * Get the bookplate names.
+     *
+     * @param $s string name of data to retrieve.
+     *
+     * @return array
+     */
+    public function getBookplateSolrData($s)
```

Comment:
I would actually move most, and maybe all, of this functionality out of the record driver and into the related record module. Most of it is only going to be used in that context, so it doesn't make sense to carry it around with every record all the time.

You can always call the getRawData() method if you want to access arbitrary fields from external code, should you wish to move getBookplateSolrData() somewhere else. As noted elsewhere, if nothing else, it doesn't make sense to have a Solr-specific method in the DefaultRecord base class, which is used by all records, not just Solr records.

Reply:
This makes sense. I've taken a stab at implementation.